### ISSUE TITLE
Algebraic definition of total array equality

### ISSUE THREAD

**Issue by daejunpark** [kframework/kore#1610]
Consider the following algebraic recursive definition of total arrays and their (partial) equality:
```
syntax BMap

syntax BMap ::= ".BMap" [function, functional, no-evaluators]

syntax Bool ::= select(BMap, Int) [function, functional]
rule select(store(M, K1, V), K2) => V             requires K1 ==Int K2
rule select(store(M, K1, _), K2) => select(M, K2) requires K1 =/=Int K2
rule select(.BMap, _) => false

syntax BMap ::= store(BMap, Int, Bool) [function, functional, no-evaluators]

syntax Bool ::= BMap "==BMap" BMap "except" Set [function, functional]

rule store(M1, K, _) ==BMap M2 except Ks
  =>       M1        ==BMap M2 except Ks
  requires K in Ks

rule M1 ==BMap store(M2, K, _) except Ks
  => M1 ==BMap       M2        except Ks
  requires K in Ks

rule M1 ==BMap M2 except _ => true
  requires M1 ==K M2  // structural equality
```

Here, the term `M ==BMap store(M, K, V) except SetItem(K)` failed to be evaluated to true.  It seems to be due to the insufficient rule priority support and the way the recursive definition is treated.

Is this a known limitation?  If so, do you have another way to define this equality?

**Comment by ttuegel** [comment]
This is a known issue (proposed to fix in #1602). You _might_ be able to work around this by giving the `[simplification]` attribute to the first two rules of `_==BMap_except_`: If the last rule (with `requires _ ==K _`) does not apply, these rules would be attempted individually.

**Comment by ttuegel** [comment]
This is resolved now that the backend supports `priority` for equations.
